### ISSUE TITLE
shouldUnregister combined with useFieldArray or dynamically visible pieces of form outside of form data.

### ISSUE THREAD

**Issue by peteristhegreat** [react-hook-form/react-hook-form#8056]
**Is your feature request related to a problem? Please describe.**
I discovered a strong solution when using these two features together, and having something in the docs about it would be nice.

**Describe the solution you'd like**
I ended up setting `shouldUnregister` to `useState` hook value, and after I read from my database and did the `reset(data)`, then I enabled `shouldUnregister`.

I guess what I am getting at is if you have dynamically mounted data, you can't have `shouldUnregister` enabled during initial mounting, but after the data comes in from the database, then you can enable it.

```
  const [allowUnregister, setAllowUnregister] = useState(false);

  const {
    register,
    handleSubmit,
    reset,
    watch,
    control,
  } = useForm({
    shouldUnregister: allowUnregister,
  });

  useEffect(() => {
    if (dbData) {
      reset(dbData);
      // other initial mount triggers based on dbData
      setAllowUnregister(true); // delaying the shouldUnregister
    }
  }, [reset, setAllowUnregister, dbData]);
```

**Describe alternatives you've considered**
I tried forcing rerenders and using `setValue` after `reset` and a whole bunch of other garbage.  I was also trying some manual trimming of data when `shouldUnregister` seemed to be causing more problems than helping.

**Additional context**
[Started with this useFieldArray example](https://jasonwatmore.com/post/2021/10/05/react-hook-form-7-dynamic-form-example-with-usefieldarray)

**Comment by bluebill1049** [comment]
hey, @peteristhegreat I am not sure if I understand correctly the feature request above. 

- you are referring to the toggle `shouldUnregister`? 
- What's use case would that fit to? 
- could you provide an example codesandbox?

Last point: I think if it's easy to do at the app level, i think it's fine with what you have showed above.

**Comment by Moshyfawn** [comment]
You can simply conditionally render the form when the data has been fetched to init your form with the actual default values from DB. Better yet, you can leverage the `React.Suspense` API.

I don't thin conditional `shouldUnregister` is the solution to your problem.

**Comment by bluebill1049** [comment]
Any update on this issue?

**Comment by peteristhegreat** [comment]
I recreated the issue here:

https://stackblitz.com/edit/react-hook-form-7-dynamic-form-example-qvn7zj?file=App.jsx

If you set `shouldUnregister` to true (line 36), then the right data shows up when you submit.  To see the issue, set the first option to false and submit... note the extra fields not visible but submitted.

But it doesn't load properly from `reset(dbData)` while `shouldUnregister` is true.

Note that when I only had the select element pointed to the fieldArray, it worked fine.  After adding the radio buttons in front of it triggered this issue for me.

If I set shouldUnregister thru a `useState` I don't have the issue.

Thanks for the patience and for looking at this.
